boring/src/ssl/mod.rs

```diff
@@ -703,31 +696,41 @@ impl SslContextBuilder {
         unsafe {
             init();
             let ctx = cvt_p(ffi::SSL_CTX_new(SslMethod::tls_with_buffer().as_ptr()))?;
+            let mut ctx = SslContextBuilder::from_ptr(ctx, true);
+
+            ctx.set_raw_public_key_mode();
 
-            Ok(SslContextBuilder::from_ptr(ctx, true))
+            Ok(ctx)
         }
     }
 
-    /// Sets raw public key certificate in DER format.
-    pub fn set_rpk_certificate(&mut self, cert: &[u8]) -> Result<(), ErrorStack> {
+    /// SSL_CTX_set_raw_public_key_mode indicates that |ctx| wishes to receive a raw
+    /// public key from a server rather than an X.509 certificate chain. Clients
+    /// should install a custom verifier with |SSL_CTX_set_custom_verify| or
+    /// |SSL_set_custom_verify| to verify the server's public key, which can be
+    /// obtained via |SSL_get0_peer_pubkey| or |SSL_get0_unparsed_peer_pubkey|. This
+    /// is only supported for client connections currently and only with TLSv1.3.
+    pub fn set_raw_public_key_mode(&mut self) {
```

Comment:
We shouldn't expose as the only right way to construct RPK is via Builde::new_rpk - it also ensures that RPK index var is set that is used to assert that RPK-incompatible API is not used